extension/oauth2clientauthextension/extension.go

```diff
@@ -38,6 +39,22 @@ type ClientCredentialsAuthenticator struct {
 // ClientCredentialsAuthenticator implements ClientAuthenticator
 var _ configauth.ClientAuthenticator = (*ClientCredentialsAuthenticator)(nil)
 
+type errorWrappingTokenSource struct {
+	ts     oauth2.TokenSource
+	config *clientcredentials.Config
```

Comment:
we just need the url - no need of having the entire config here. Please adjust the struct definition.